brainglobe_stitch/file_utils.py

```diff
@@ -226,3 +226,108 @@ def get_slice_attributes(
         slice_attributes[name] = sub_dict
 
     return slice_attributes
+
+
+def get_big_stitcher_transforms(
+    xml_path: Path, x_size: int, y_size: int, z_size: int
+) -> List[List[int]]:
+    """
+    Get the translations for each tile from a Big Data Viewer XML file.
+    The translations are calculated by BigStitcher.
+
+    Parameters
+    ----------
+    xml_path: Path
+        The path to the Big Data Viewer XML file.
+    x_size: int
+        The size of the image in the x-dimension.
+    y_size: int
+        The size of the image in the y-dimension.
+    z_size: int
+        The size of the image in the z-dimension.
+
+    Returns
+    -------
+    List[List[int]]
+        A list of translations for each tile. Each translation is a list with
+        the following order: [x_start, x_end, y_start, y_end, z_start, z_end].
+    """
+    tree = ET.parse(xml_path)
+    root = tree.getroot()
+
+    stitch_transforms = root.findall(
+        ".//ViewTransform/[Name='Stitching Transform']/affine"
+    )
+    assert (
```

Comment:
To be honest this was mostly here to satisfy type checking by `mypy`, the `findall` method returns a `Union[str, None, any]` and `mypy` doesn't like when I try to access a field within the object later down in the code. The alternative was putting `if` statements checking for `None` for each field and I honestly found the `assert` statements cleaner. However, on second though I might actually go back to `if` statements that raise errors instead!

Reply:
That makes sense, I didn't clock fully that these were all checks for `None`. Yes that's annoying, I wonder, if the likelihood of the underlying data actually being `None` is very low, then maybe the messages could be something like `Typenarrow `variable_name`', rather than user-facing, (though maybe this is confusing in other ways). Alternatively you could introduce a function like 
```
typesafe_findall(root, str):
x = root.findall(str)
assert x is not `None`
return x
```
but this pattern isn't used that much, it adds indirection, and it doesn't solve the `x[0].text` typing nor the `root.find` case, so maybe not worth it. Maybe changing the assert message (if appropriate) could work better.

Reply:
I like the idea of a separate function, that at least confines all the messy code to one area and leaves the rest looking nice!